### PR TITLE
Implement SG workaround in Public cloud

### LINKED ISSUE #68: Load Balancer Service become unavailable in public cloud
**What happened**:

We have a limitation related to IaaS's Load Balancer Unit (LBU) when created in a public cloud context.

All LBU of a specific cloud account which are created in public cloud use the same Security Group (SG) "outscale-elb-sg" which cannot be controlled by the user. See more details in [Load Balacer Unit documentation regarding security groups](https://docs.outscale.com/en/userguide/About-Security-Groups-in-LBU.html).

When a Service of type LoadBalancer is created in Kubernetes, the CCM dynamically add a rule in the SG which has the  "OscK8sMainSG" tag. This SG is applied to all workers.
The added rule allow all trafic from members of "outscale-elb-sg" SG (all account's LBU created in the public cloud).

When a Service of type LoadBalancer is deleted, this rule is automatically deleted from "OscK8sMainSG" tagged SG.
This behavior is problematic when a worker node still need to accept traffic from another LBU which is part of a separate Service of type LoadBalancer.

This behavior is specific to an LBU created in public cloud as this SG cannot be chosen, cannot be edited and is unique.

**What you expected to happen**:

When used in public cloud, deleting a Service of type LoadBalancer does not affect other LoadBalancer Services hosted on the same nodes.

**How to reproduce it (as minimally and precisely as possible)**:

- Have a cluster hosted on public cloud with CCM installed with 1 worker.
- Create a Service A of type LoadBalancer
- Service A is available
- Create a Service B of type LoadBalancer
- Service B is available
- Delete Service B
- Service B is not available
- Service A is not available 🔥 

**Workaround**

1. Create a new SG
2. Add an Inbound rule allowing all trafic (protocol -1) from all members of "outscale-elb-sg" SG
3. Apply this SG to all worker nodes

This should not be a security problem as LBU's trafic in routed to specific backends and trafic only originate from those LBU.
However, this would always be better to not have this rule in the first place.

**Resolution Notes**

- For dev: be sure to reference this issue in code
- Need additional documentation/reference to official documentation regarding public cloud LBU behavior
- Add test for this scenario

/kind bug

### PR DESCRIPTION
**Pull request by outscale-mdr**
Close #68

Improve all testing cases will be done in #128

### CODE REVIEW

**Comment by outscale-mdr** [comment]
Need outscale-dev/osc-k8s-rke-cluster/pull/14

**Comment by outscale-mdr** [comment]
Proof of success of the GH Action: https://github.com/outscale-mdr/cloud-provider-osc/runs/8211497738?check_suite_focus=true

File: tests/e2e/lb_crud.go [hunk 1]
```diff
@@ -179,3 +179,107 @@ var _ = ginkgo.Describe("[ccm-e2e] SVC-LB", func() {
...
+		ginkgo.By("listDeployment")
+		e2eutils.ListDeployment(cs, nsSvc1)
+		e2eutils.ListDeployment(cs, nsSvc2)
+
+		ginkgo.By("Create Svc 1")
+		svc1 := e2eutils.CreateSvc(cs, nsSvc1, annotations)
+		fmt.Printf("Created Service %q.\n", svc1)
+		defer e2eutils.ListSvc(cs, nsSvc1)
+
+		ginkgo.By("Create Svc 1")
```
Comment (jerome-jutteau) [inline tests/e2e/lb_crud.go:229]: > Create Svc 1
to
> Create Svc 2

**Review by jerome-jutteau — CHANGES_REQUESTED** [review]

File: tests/e2e/lb_crud.go (hunk 1 above)
Comment (outscale-mdr) [reply tests/e2e/lb_crud.go:229]: Thank you

**Comment by outscale-mdr** [comment]
CI build failes due to CVE not code

**Review by jerome-jutteau — APPROVED** [review]

**Pull request closed without merging**